docs/lab/lab04/lab04.md

```diff
@@ -0,0 +1,384 @@
+---
+sidebar_position: 5
+description: PWM & ADC
+---
+
+# 04 - PWM & ADC
+
+This lab will teach you the difference between digital and analog signals, how to simulate analog signals by using Pulse Width Modulation (PWM) and how to convert analog signals to digital ones using Analog-to-Digital Converters (ADC).
+
+
+## Resources
+
+TBD
+
+## Timing
+
+### Clocks
+
+In embedded applications, keeping track of time is crucial. Even for the simple task of blinking a led at a certain time interval, we need a reference of time that is constant and precise. 
+A clock is a piece of hardware that provides us with that reference. Its purpose is to oscillate at a fixed frequency and provide a signal that switches from high to low at a fixed interval.
+
+![ClockSignal](images/clock_signal.png)
+
+The most precise type of clock is the crystal oscillator (XOSC). The reason why it is so accurate is because it uses the crystal's natural vibration frequency to create the clock signal. This clock is usually external to the processor itself, but the processor also has an internal clock (ROSC) that is less accurate and that can be used in cases where small variations of clock  pulses are negligeable. When using the USB protocol, for instance, a more stable clock signal is required, therefore the XOSC is necessary.
```

Comment:
What about adding a picture similar to https://kagi.com/images?q=raspberry+pi+pico+crystal, but that only highlights the crystal.

Reply:
Done